src/components/Form/SetAclPermissionForm.jsx

```diff
@@ -109,51 +76,80 @@ const SetAclPermissionForm = () => {
+              id="permissionType"
+              label="Select One"
+              value={permissionState.permissionType}
+              onChange={(e) =>
+                setPermissionState({ ...permissionState, permissionType: e.target.value })
+              }
+              name="setAclPerms"
+            >
+              <MenuItem value="Give">Give Permission</MenuItem>
+              <MenuItem value="Revoke">Revoke Permission</MenuItem>
+            </Select>
+          </FormControl>
+          <br />
+          <FormControl fullWidth sx={{ marginBottom: '1rem' }}>
             <TextField
               id="set-acl-to"
               name="setAclTo"
-              value={username}
-              onChange={(e) => setUsername(e.target.value)}
-              placeholder={username}
-              label="Search Username"
+              value={permissionState.podUrlToSetPermissionsTo}
+              onChange={(e) =>
+                setPermissionState({ ...permissionState, podUrlToSetPermissionsTo: e.target.value })
+              }
+              placeholder={permissionState.podUrlToSetPermissionsTo}
+              label="Enter podURL"
               required
+              error={permissionState.podUrlToSetPermissionsTo === podUrl}
+              helperText={
+                permissionState.podUrlToSetPermissionsTo === podUrl
+                  ? 'Cannot modify your permissions to your own pod.'.toUpperCase()
+                  : ''
+              }
             />
           </FormControl>
+
           <DocumentSelection
             htmlForAndIdProp="set-acl-doctype"
             handleDocType={handleDocType}
             docType={docType}
           />
-          <br />
-          <FormControl fullWidth>
-            <FormLabel htmlFor="set-acl-perm-label">Select permission setting:</FormLabel>
-            <RadioGroup row aria-labelledby="set-acl-perm-label" name="set-acl-perm">
-              <FormControlLabel
-                value="Give"
-                control={<Radio />}
-                label="Give"
-                id="set-acl-perm-give"
-                name="setAclPerms"
-              />
-              <FormControlLabel
-                value="Revoke"
-                control={<Radio />}
-                label="Revoke"
-                id="set-acl-perm-revoke"
-                name="setAclPerms"
-              />
-            </RadioGroup>
-            <Button variant="contained" disabled={state.processing} type="submit" color="primary">
-              Set Permission
+
+          <FormControl fullWidth sx={{ marginTop: '2rem' }}>
+            <Button
+              variant="contained"
+              disabled={permissionState.podUrlToSetPermissionsTo === podUrl || state.processing}
+              type="submit"
+              color="primary"
+            >
+              {permissionState.permissionType
+                ? `${permissionState.permissionType} Permission`
+                : 'Give or Revoke Permission'}
```

Comment:
I think we could leave the original text as `Set Permission` and have the word "Set" changed into "Give" or "Revoke" rather than `Give or Revoke`.